ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot

```diff
@@ -89,27 +95,35 @@ Install RHODS
   END
   Clone OLM Install Repo
   Configure Custom Namespaces
-  IF   "${PRODUCT}" == "ODH"
-       ${csv_display_name} =    Set Variable    ${ODH_CSV_DISPLAY}
-  ELSE
-       ${csv_display_name} =    Set Variable    ${RHODS_CSV_DISPLAY}
-  END
   IF   "${cluster_type}" == "selfmanaged"
       IF  "${TEST_ENV}" in "${SUPPORTED_TEST_ENV}" and "${INSTALL_TYPE}" == "Cli"
              Install RHODS In Self Managed Cluster Using CLI  ${cluster_type}     ${image_url}
       ELSE IF  "${TEST_ENV}" in "${SUPPORTED_TEST_ENV}" and "${INSTALL_TYPE}" == "OperatorHub"
-          ${file_path} =    Set Variable    tasks/Resources/RHODS_OLM/install/
-          Copy File    source=${file_path}cs_template.yaml    destination=${file_path}cs_apply.yaml
-          IF  "${PRODUCT}" == "ODH"
-              Run    sed -i'' -e 's/<CATALOG_SOURCE>/community-operators/' ${file_path}cs_apply.yaml
-          ELSE
-              Run    sed -i'' -e 's/<CATALOG_SOURCE>/${CATALOG_SOURCE}/' ${file_path}cs_apply.yaml
+          IF  "${is_upgrade}" == "False"
+              ${file_path} =    Set Variable    tasks/Resources/RHODS_OLM/install/
+              ${starting_csv} =  Set Variable    ${EMPTY}
+              IF  "${rhoai_version}" != "${EMPTY}"
+                  Log    Start installing "${OPERATOR_NAME}" with version: ${rhoai_version}    console=yes
+                  ${starting_csv} =  Set Variable    ${OPERATOR_DEPLOYMENT_NAME}.${rhoai_version}
+              END
+              ${destination_file} =    Set Variable    ${file_path}cs_apply.yaml
+              Copy File    source=${file_path}cs_template.yaml    destination=${destination_file}
+              Run    sed -i'' -e 's/<CATALOG_SOURCE>/${CATALOG_SOURCE}/' ${destination_file}
+              Run    sed -i'' -e 's/<OPERATOR_NAME>/${OPERATOR_DEPLOYMENT_NAME}/' ${destination_file}
+              Run    sed -i'' -e 's/<OPERATOR_NAMESPACE>/${OPERATOR_NAMESPACE}/' ${destination_file}
+              Run    sed -i'' -e 's/<UPDATE_CHANNEL>/${UPDATE_CHANNEL}/' ${destination_file}
+              Run    sed -i'' -e 's/<STARTING_CSV>/${starting_csv}/' ${destination_file}
```

Comment:
iirc you cannot pass the startingCSV as empty or OLM will fail. Have you tested it?

Reply:
the code still allows to startingCSV to be empty. Have you tried that case? Does it work?